pkg/porter/publish.go

```diff
@@ -0,0 +1,172 @@
+package porter
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/deislabs/porter/pkg/config"
+	"github.com/docker/cli/cli/command"
+	cliflags "github.com/docker/cli/cli/flags"
+	"github.com/docker/distribution/reference"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/pkg/jsonmessage"
+	"github.com/docker/docker/pkg/term"
+	"github.com/docker/docker/registry"
+	"github.com/pkg/errors"
+)
+
+// PublishOptions are options that may be specified when publishing a bundle.
+// Porter handles defaulting any missing values.
+type PublishOptions struct {
+	File     string
+	Tag      string
+	CNABFile string
+}
+
+func (p PublishOptions) Validate(porter *Porter) error {
+	if p.File != "" {
+		return nil
+	}
+	fs := porter.Context.FileSystem
+	pwd, err := os.Getwd()
+	if err != nil {
+		return errors.Wrap(err, "could not get current working directory")
+	}
+	fmt.Println(pwd)
+	files, err := fs.ReadDir(pwd)
```

Comment:
Why not use `fs.Exists` to check if porter.yaml is in the current directory?

Reply:
I copied this from the existing validation code (that wasn't directly reusable). I'll update.